packages/did/src/Did.signature.ts

```diff
@@ -54,32 +57,28 @@ function verifyDidSignatureDataStructure(
  *
  * @param input Object wrapping all input.
  * @param input.message The message that was signed.
- * @param input.signature An object containing signature and signer key.
+ * @param input.signature Signature bytes.
+ * @param input.keyUri DID URI of the key used for signing.
  * @param input.expectedVerificationMethod Which relationship to the signer DID the key must have.
  * @param input.didResolveKey Allows specifying a custom DID key resolve. Defaults to the built-in [[resolveKey]].
  */
 export async function verifyDidSignature({
   message,
   signature,
+  keyUri,
   expectedVerificationMethod,
   didResolveKey = resolveKey,
 }: DidSignatureVerificationInput): Promise<void> {
-  verifyDidSignatureDataStructure(signature)
-  // Add support for old signatures that had the `keyId` instead of the `keyUri`
-  const inputUri = signature.keyUri || (signature as any).keyId
   // Verification fails if the signature key URI is not valid
-  const { fragment } = parse(inputUri)
+  const { fragment } = parse(keyUri)
   if (!fragment)
     throw new SDKErrors.SignatureMalformedError(
-      `Signature key URI "${inputUri}" invalid`
+      `Signature key URI "${keyUri}" invalid`
     )
 
-  const { publicKey } = await didResolveKey(
-    inputUri,
-    expectedVerificationMethod
-  )
+  const { publicKey } = await didResolveKey(keyUri, expectedVerificationMethod)
 
-  Crypto.verify(message, signature.signature, u8aToHex(publicKey))
+  Crypto.verify(message, signature, u8aToHex(publicKey))
```

Comment:
Why doesn't `Crypto.verify()` also take a `Uint8Array`? I thought we wanted to get rid of "useless" conversions between HEX string and byte arrays.

Reply:
At the end a raw public key is represented by bytes, not by a HEX string.

Reply:
I know it's out of scope for this PR, but this could potentially be another breaking change.

Reply:
you're right, that could be fixed too